tests/test_dir_traversal.py

```diff
@@ -106,8 +128,17 @@ class DirectoryTraversalVector(Generic[T]):
     given: List[BaseEntry]
     add: List[AddCall] = field(default_factory=lambda: [AddCall(path=".")])
     skip_dirs: List[str] = field(default_factory=list)
+    ignore_patterns: List[IgnorePattern] = field(default_factory=list)
+    ignore_exceptions: List[Type[Exception]] = field(default_factory=list)
+    ignore_patterns_raise_ValueError: bool = False
     expect: List[ExpectedTraverseStep] = field(default_factory=list)
     expect_alternatives: Optional[List[List[ExpectedTraverseStep]]] = None
+    skip_me: Callable[[], Optional[str]] = lambda: None
```

Comment:
Maybe we could simplify the type here to just:
```suggestion
    skip_me: Optional[str] = None
```
We use only `on_windows` function to pupolate this field for now, and maybe we do not need to over-abstarct this.
Unless there is some other reason, that skips me.

Reply:
True, I was thinking _too_ generically here (that the test vector should be able to check something _when the test is run_ to decide whether or not to skip), but (since Python is an interpreted language) the Windows-or-not question is already known when we create the test vector here. Will fix. 🙏🏻

Reply:
Fixed in 584fc1d45a1b570fd414e74b4b8cf3539674978c.